pkg/kubectl/cmd/util/factory_builder.go

```diff
@@ -84,3 +88,50 @@ func (f *ring2Factory) PluginLoader() plugins.PluginLoader {
 func (f *ring2Factory) PluginRunner() plugins.PluginRunner {
 	return &plugins.ExecPluginRunner{}
 }
+
+func (f *ring2Factory) ScaleClient() (scaleclient.ScalesGetter, error) {
+	discoClient, err := f.clientAccessFactory.DiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	restClient, err := f.clientAccessFactory.RESTClient()
+	if err != nil {
+		return nil, err
+	}
+	resolver := scaleclient.NewDiscoveryScaleKindResolver(discoClient)
+	mapper, _ := f.objectMappingFactory.Object()
+	return scaleclient.New(restClient, mapper, dynamic.LegacyAPIPathResolverFunc, resolver), nil
+}
+
+func (f *ring2Factory) Scaler(mapping *meta.RESTMapping) (kubectl.Scaler, error) {
+	clientset, err := f.clientAccessFactory.ClientSet()
+	if err != nil {
+		return nil, err
+	}
+
+	scalesGetter, err := f.ScaleClient()
+	if err != nil {
+		return nil, err
+	}
+	gvk := mapping.GroupVersionKind.GroupVersion().WithResource(mapping.Resource)
+
+	return kubectl.ScalerFor(mapping.GroupVersionKind.GroupKind(), clientset.Batch(), scalesGetter, gvk.GroupResource()), nil
+}
+
+func (f *ring2Factory) Reaper(mapping *meta.RESTMapping) (kubectl.Reaper, error) {
+	clientset, clientsetErr := f.clientAccessFactory.ClientSet()
+	scaler, err := f.ScaleClient()
+	if err != nil {
+		return nil, err
+	}
+	reaper, reaperErr := kubectl.ReaperFor(mapping.GroupVersionKind.GroupKind(), clientset, scaler)
+
+	if kubectl.IsNoSuchReaperError(reaperErr) {
+		return nil, reaperErr
+	}
+	if clientsetErr != nil {
```

Comment:
Move this check right after `f.clientAccessFactory.ClientSet()`. What's the point of continuing if we failed getting clientset. Generally fail fast is the desired approach.

Reply:
I see it's copy&paste, but sine you've touched it, can you please fix it :wink:

Reply:
sure, no problem.